pkg/upload/react-native-cocoa.go

```diff
@@ -0,0 +1,195 @@
+			projectRoot = rootDirPath
+		}
+
+		// Check if we're missing any of these parameters
+		if bundlePath == "" || plistPath == "" && (apiKey == "" || versionName == "" || bundleVersion == "") {
+
+			// Check to see if we have the xcworkspacePath
+			if xcworkspacePath == "" {
+				// If not, attempt to locate it in the path/ios/ folder
+				if utils.IsDir(filepath.Join(path, "ios")) {
+					workspacePath, err := utils.FindFolderWithSuffix(filepath.Join(path, "ios"), ".xcworkspace")
+					if err != nil {
+						return err
+					}
+					xcworkspacePath = workspacePath
+				} else {
+					return fmt.Errorf("unable to find xcworkspace file in your project, please specify the path using --xcworkspace")
+				}
+
+			}
+
+			// Check to see if we have a scheme
+			if scheme == "" {
+				// If not, work it out from the xcworkspace file
+				possibleSchemeName := strings.TrimSuffix(filepath.Base(xcworkspacePath), ".xcworkspace")
+				schemeExists, err := cocoa.IsSchemeInWorkspace(xcworkspacePath, possibleSchemeName)
+				if err != nil {
+					return err
+				}
+
+				if schemeExists {
+					scheme = possibleSchemeName
+				}
+			}
+
+			// Pull build settings from the xcworkspace file
+			var err error
+			buildSettings, err = cocoa.GetXcodeBuildSettings(xcworkspacePath, scheme)
+			if err != nil {
+				return err
+			}
+
+			// Check to see if we have the Info.Plist path
+			if plistPath == "" {
+				// If not, we need to build it from build settings values
+				plistPathExpected := filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.InfoPlistPath)
+				if utils.FileExists(plistPathExpected) {
+					plistPath = plistPathExpected
+					log.Info("Found Info.plist at: " + plistPath)
+				} else {
+					log.Info("No Info.plist found at: " + plistPathExpected)
+				}
+			}
+		}
+
+		// Set a sourceMapPath if it's not defined and check that it exists before proceeding
+		if sourceMapPath == "" {
+			// If this environment variable is set, use it to override default path 'buildDirPath'
+			sourceMapFileEnvVar := os.Getenv("SOURCEMAP_FILE")
+			if sourceMapFileEnvVar != "" {
+				buildDirPath = sourceMapFileEnvVar
+			}
+			sourceMapPath = filepath.Join(buildDirPath, "sourcemaps", "main.jsbundle.map")
+			if !utils.FileExists(sourceMapPath) {
+				return errors.New("Could not find a suitable source map file, please specify the path by using --source-map")
+			}
+		}
+
+		// Set a bundlePath if it's not defined and check that it exists before proceeding
+		if bundlePath == "" {
+			bundleFilePath := filepath.Join(buildSettings.ConfigurationBuildDir, "main.jsbundle")
+			if !utils.FileExists(bundleFilePath) {
+				return errors.New("Could not find a suitable bundle file, please specify the path by using --bundlePath")
+			}
+			bundlePath = bundleFilePath
+		}
+
+		if plistPath != "" && (apiKey == "" || versionName == "" || bundleVersion == "") {
+			// Read data from the plist
+			plistData, err := cocoa.GetPlistData(plistPath)
```

Comment:
So it looks like it will just skip on to the next step and error about there being no source map:
`[ERROR] Could not find a suitable source map file, please specify the path by using --source-map`
I think a better condition would be to do something like `if plistPath == "" || !utils.FileExists(plistPath)` on line 114, that way we could potentially find it via `plistPathExpected := filepath.Join(buildSettings.ConfigurationBuildDir, buildSettings.InfoPlistPath)` and if that still fails, then we'd automatically log `log.Info("No Info.plist found at: " + plistPathExpected)`, if that seems feasible? That way the user can see that 1) There's no Info.plist found 2) Which has resulted in not finding a source map

Reply:
Also just to note, I don't think we can reach line 150 with a non-existant plist. It will just state the above, about not being able to find the source map file

Reply:
Hmm - I just wonder if you run the risk of confusing the user if they provide one plist file and we locate and use a different one?

IMO, we should simply error out if the user provides a path that doesn't exist when we come to use it. Happy to discuss if you feel differently though.